[clang-tidy][modernize-use-starts-ends-with] Add support for two ends_with patterns

nicovank opened the pull request:
Add support for the following two patterns:
```
haystack.compare(haystack.length() - needle.length(), needle.length(), needle) == 0;
haystack.rfind(needle) == (haystack.size() - needle.size());
```

nicovank commented:
Oops, for this reason my software duplicated #105366 when I meant to just rebase and ping.
Sorry. Closing the old one.

5chmidti reviewed (approved):
LGTM

(got ideas on writing a small part of the matchers differently for an NFC I'll do afterward)

nicovank commented:
Sounds good, thanks! I will go ahead and rebase and merge this.

nicovank committed:
[clang-tidy][modernize-use-starts-ends-with] Add support for two ends_with patterns

Add support for the following two patterns:
```
haystack.compare(haystack.length() - needle.length(), needle.length(), needle) == 0;
haystack.rfind(needle) == (haystack.size() - needle.size());
```
